go-controller/pkg/libovsdb/util/router_test.go

```diff
@@ -0,0 +1,99 @@
+package util
+
+import (
+	"fmt"
+	"net"
+	"testing"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	libovsdbtest "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/testing/libovsdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+)
+
+func TestCreateDefaultRouteToExternal(t *testing.T) {
+	nodeName := "ovn-control-plane"
+	_, cidr4, _ := net.ParseCIDR("10.128.0.0/14")
+	config.Default.ClusterSubnets = []config.CIDRNetworkEntry{{cidr4, 24}}
+	expectedStaticRoute := &nbdb.LogicalRouterStaticRoute{
+		Nexthop:  "100.64.0.1",
+		IPPrefix: "10.128.0.0/14",
+		Policy:   &nbdb.LogicalRouterStaticRoutePolicySrcIP,
+		UUID:     "route1-UUID",
+	}
+	unexpectedStaticRoute := &nbdb.LogicalRouterStaticRoute{
+		Nexthop:  "100.66.0.1",
+		IPPrefix: "10.128.0.0/14",
+		Policy:   &nbdb.LogicalRouterStaticRoutePolicySrcIP,
+		UUID:     "route2-UUID",
+	}
+	initialOVNClusterRouter := &nbdb.LogicalRouter{
+		UUID:         types.OVNClusterRouter + "-UUID",
+		Name:         types.OVNClusterRouter,
+		StaticRoutes: []string{"route2-UUID"},
+	}
+	expectedOVNClusterRouter := &nbdb.LogicalRouter{
+		UUID:         types.OVNClusterRouter + "-UUID",
+		Name:         types.OVNClusterRouter,
+		StaticRoutes: []string{"route1-UUID"},
+	}
+	expectedLogicalRouterPort := &nbdb.LogicalRouterPort{
+		Name:     types.GWRouterToJoinSwitchPrefix + types.GWRouterPrefix + nodeName,
+		UUID:     types.GWRouterToJoinSwitchPrefix + types.GWRouterPrefix + nodeName + "-UUID",
+		Networks: []string{"100.64.0.1/16"},
+	}
+	GRName := "GR_" + nodeName
+	expectedOVNGatewayRouter := &nbdb.LogicalRouter{
+		UUID:  GRName + "-UUID",
+		Name:  GRName,
+		Ports: []string{types.GWRouterToJoinSwitchPrefix + types.GWRouterPrefix + nodeName + "-UUID"},
+	}
+
+	tests := []struct {
+		desc         string
+		initialNbdb  libovsdbtest.TestSetup
+		expectedNbdb libovsdbtest.TestSetup
+	}{
+		{
+			desc: "removes stale static route and creates new route with current gateway router logical port IP",
```

Comment:
Thanks for adding the test!!
Would you mind adding one for the first commit too?